File: 01_Language/01_Functions/php/get_parent_class.php
<?php

class dad {
    function __construct()
    {
        // implements some logic
    }
}

class child extends dad {
    function __construct()
    {
        echo "I'm " , get_parent_class($this) , "'s son\n";
    }
}

class child2 extends dad {
    function __construct()
    {
        echo "I'm " , get_parent_class('child2') , "'s son too\n";
    }
}

$foo = new child();
$bar = new child2();

File: 01_Language/01_Functions/python/get_parent_class.py
# coding: utf-8

def get_parent_class(cls):
    if not isinstance(cls, type):
        cls = type(cls)
    return cls.__bases__


if __name__ == '__main__':
    class dad(object):
        pass


    class child(dad):
        def __new__(cls, *args, **kwargs):
            print(cls, args, kwargs)
            print("I'm ", get_parent_class(cls), "'s son")
            return object.__new__(cls)


    class child2(dad):
        def __new__(cls, *args, **kwargs):
            print(cls, args, kwargs)
            o = object.__new__(cls)
            print("I'm ", get_parent_class(o), "'s son too")
            return o


    f = child()
    b = child2()
